### PR TITLE
Sign-up Feature

### PR DESCRIPTION
**Pull request by songrenzhao**
### Overview of PR
- [x] Finalized with our sign-up logic  
- [ ] Finishing up token verification logic 
### Architecture Changes
- [x] Added signup routes, ready to use
  - [x] `{host}/verify` => Will be change to `{host}/signup`
- [x] Added token verification service
  - [ ] Will add route soon
### Architecture Changes
 - Looking to launch CD soon, please look into it @jinchen1036 
### ToDo
 - Secure endpoints with JWT

### CODE REVIEW

File: src/services/signUpService.js [hunk 1]
```diff
@@ -53,10 +54,11 @@ export async function registerUserInfo(userInfo) {
     if (userExist) {
       throw new Error('This email or username has already been taken');
     }
+    const hashedPassword = await bcrypt.hash(password, 12);
```
Comment (songrenzhao) [inline src/services/signUpService.js]: Does this have a unhash function?

Comment (JiejayLan) [reply src/services/signUpService.js]: means hash the regular password into a long string, for example:$2a$12$gr9q5FcjVruD9OOtOiv3/efZvJJpX6DeM/R8fOnKvEK6J224wlGEe

File: src/services/__tests__/signUpService.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+import {
+  sendVerificationEmail, registerVerificationToken,
+  registerUserInfo, signUpAccount,
+} from '../signUpService';
+import model from '../../models';
+
+jest.mock('../../models');
+jest.mock('@sendgrid/mail');
+
+describe('App test suite', () => {
```
Comment (JiejayLan) [inline src/services/__tests__/signUpService.js]: correct the description message

File: src/services/__tests__/signUpService.js [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+  beforeEach(() => {
+    sgMail.send.mockClear();
+    model.db.VerificationToken.create.mockClear();
+    model.db.Users.create.mockClear();
+    model.db.Users.findOne.mockClear();
+  });
+
+  describe('registerVerificationToken test suite', () => {
+    it('Should call service message', async () => {
+      const result = model.db.VerificationToken.create.mockReturnValue({});
```
Comment (JiejayLan) [inline src/services/__tests__/signUpService.js]: I will to discuss with you about how necessary to check if result is undefined here

**Pull request merged**